actionview/lib/action_view/helpers/tags/datetime_local_field.rb

```diff
@@ -16,6 +16,14 @@ def field_type
         end
 
         private
+          def normalize_datetime(value)
+            parsed = parse_datetime(value)
+
+            return value if parsed&.strftime("%Y-%m-%dT%H:%M") == value
```

Comment:
This seems rigid to me.  Also a bit wasteful, performance-wise.

Since the format of an `options["value"]` string can affect the presentation, I think we should just pass it through as is.  In other words, if the user wants to specify an exact string for `options["value"]`, we should let them.

(Though I don't think it is the same for `options["min"]` or `options["max"]`, since those are not graphically rendered.)

I've pushed a commit to show what I mean.  Let me know what you think.  If it looks good, you or I can squash.

Reply:
I like your version much better! It's what I suggested here after dwelling on this PR more: https://github.com/rails/rails/pull/46678#issuecomment-1496812169

Reply:
I've squashed :)

Reply:
Oh, right you are!  Sorry, I had bookmarked this PR, but forgot to go back and read your comment.  But I definitely agree! :smile: